### ISSUE TITLE
Dependencies are causing problems for pip install

### ISSUE THREAD

**Issue by HoniSanders** [AttackIQ/SigmAIQ#8]
It seems that `sigmaiq` requires `langchain-community` even though it is not listed as a dependency.  For example:
```
from sigmaiq.llm.base import SigmaLLM
```

gives :  
```
ModuleNotFoundError: Module langchain_community.embeddings not found. Please install langchain-community to access this module. You can install it using `pip install -U langchain-community`
```

However, you cannot install both at the same time.  
```
pip install langchain-community sigmaiq
```
takes forever to resolve and gives 
```
ERROR: Cannot install langchain-community and sigmaiq because these package versions have conflicting dependencies.

The conflict is caused by:
    langchain-core 0.0.13 depends on packaging<24.0 and >=23.2
    pysigma 0.9.11 depends on packaging<23.0 and >=22.0
```

So, can you fix two problems:
- Include `langchain_community` as a dependency
- Fix the dependency conflicts

Thank you

**Comment by HoniSanders** [comment]
It seems that upgrading `pysigma` to v0.11 solves these dependency problems.  Can you allow that upgrade in your dependencies?  It seems that you currently require `pysigma` exactly equal to `0.10.10`.

**Comment by slincoln-aiq** [comment]
Hey there,

Sorry for not getting back to you sooner. I released v0.4.2 last week which, among other features/fixes, should fix this issue.  Please upgrade to the new version and let me know if you still run into the same issue.

Thanks!
